src/time-picker/time-picker.jsx

```diff
@@ -109,6 +109,17 @@ const TimePicker = React.createClass({
     };
   },
 
+  componentWillReceiveProps(nextProps, nextContext) {
+    const newState = this.state;
+    if (nextContext.muiTheme) {
+      newState.muiTheme = nextContext.muiTheme;
+    }
+    if (!Date.isEqualTime(this.state.time, nextProps.defaultTime)) {
+      newState.time = nextProps.defaultTime;
```

Comment:
I think that the `defaultTime` should only be used during the first render, like for a regular `<input />`. I would add a new `value` property for your "controlled" use case.`
Implementing this would also be a good occasion to deprecate this ugly imperative `setTime()` method that is used for our `Controlled examples` here http://0.0.0.0:3000/#/components/time-picker.